app/controllers/verify/finance_controller.rb

```diff
@@ -7,13 +7,18 @@ class FinanceController < ApplicationController
     before_action :confirm_step_allowed
     before_action :submit_idv_form, only: [:create]
     before_action :submit_idv_job, only: [:create]
+    before_action :refresh_if_not_ready, only: [:show]
```

Comment:
Ok so I tried extracting into a controller, and due to the mixins we use and the amount of shared methods between the different views (ex `step` and `idv_form`), it did not separate cleanly. I was able to remove most of the `before_action`s by integrating them, so I'm pushing that change up now: f627bc703f8fce506dbd7943b51de09e234d002b